frontend/public/components/environment.jsx

```diff
@@ -80,6 +80,149 @@ const stateToProps = ({k8s}, {obj}) => ({
   model: k8s.getIn(['RESOURCES', 'models', referenceFor(obj)]) || k8s.getIn(['RESOURCES', 'models', obj.kind]),
 });
 
+class CurrentEnvVars {
+  constructor() {
+    this.currentEnvVars = {};
+  }
+
+  instantiateWithRawData(rawEnvData) {
+    this.isContainerArray = _.isArray(rawEnvData.containers);
+    this.hasInitContainers = !_.isUndefined(rawEnvData.initContainers);
+
+    if (this.isContainerArray) {
+      this.currentEnvVars.containers = envVarsToArray(rawEnvData.containers);
+      this.currentEnvVars.initContainers = envVarsToArray(rawEnvData.initContainers);
+    } else {
+      this.currentEnvVars.buildObject = envVarsToArray(rawEnvData);
+    }
+    return this;
+  }
+
+  instantiateWithResultObject(resultObject, isContainerArray, path) {
+    if (isContainerArray) {
+      return this.instantiateWithRawData(_.get(resultObject, _.dropRight(path)));
+    }
+    return this.instantiateWithRawData(_.get(resultObject, path));
+  }
+
+  getEnvVarByTypeAndKey(containerType, key) {
+    if (this.isContainerArray) {
+      return this.currentEnvVars[containerType][key];
+    }
+    return this.currentEnvVars.buildObject[key];
+  }
+
+  setFormattedVars(containerType, index, environmentType, formattedPairs) {
```

Comment:
What are "formatted" vars?

Reply:
This is data that is formatted for the name value editor vs the passed in object data.